### PR TITLE
Update s3 tests to use moto_server

### PR DESCRIPTION
**Pull request by ayushdg**
This PR updates s3 tests to use `moto_server` instead of going via a moto mock_s3 context. This enables cleaner s3 testing with `s3fs>=0.5` which incorporates aiobotocore for s3 connections.

- The pytests starts up a moto-server for each worker running tests. 
- Ports used: `5000, 5550 - 5550+ (n_pytest_workers-1)`

Updated integration repo with requirements: https://github.com/rapidsai/integration/pull/207

### CODE REVIEW

File: python/cudf/cudf/tests/test_s3.py
Comment (ayushdg) [inline python/cudf/cudf/tests/test_s3.py:51]: This approach potentially fails if pytest is run on more than 10 workers.

Comment (kkraus14) [reply python/cudf/cudf/tests/test_s3.py:51]: Can we do integer math and then cast the integer to a string?

**Comment by ayushdg** [comment]
Failing dask-cudf test attributed to changes here: https://github.com/dask/dask/pull/6960#discussion_r557843873.
Will rerun tests once fix goes into dask.

**Comment by jrbourbeau** [comment]
@rjzamora's fix in https://github.com/dask/dask/pull/7074 has been merged

**Comment by ayushdg** [comment]
rerun tests

File: ci/gpu/build.sh
Comment (ayushdg) [inline ci/gpu/build.sh:86]: @kkraus14  Would the next step be to incorporate these changes to https://github.com/rapidsai/integration, revert them here and only merge this PR once the integration is merged?

Comment (kkraus14) [reply ci/gpu/build.sh:86]: Yup, exactly

**Review by galipremsagar — APPROVED** [review]

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**
